test/fail_compilation/test18295.d

```diff
@@ -0,0 +1,24 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/test18295.d(14): Error: functions cannot return `scope test18295.C`
+---
+*/
+
+// https://issues.dlang.org/show_bug.cgi?id=18295
+
+// See test/compilable/test18295.d for the `-dip1000` version`
+
+scope class C { int i; }    // Notice the use of `scope` here
+
+C increment(C c)            // Not compiling under -dip1000, so this should emit an error
+{
+    c.i++;
+    return c;
+}
+
+void main()
+{
+    scope C c = new C();
+    c.increment();
```

Comment:
That simply won't catch the escaping of a scope variable, checks are only enabled with `-dip1000`.

Reply:
That's not the point of this test.  The point is to test the current behavior for which there currently is no test.  See https://codecov.io/gh/dlang/dmd/src/master/src/dmd/typesem.d#L800...801  If you believe that is an error, please file a bug in bugzilla and submit a fix in a separate PR.